crates/bin/src/args.rs

```diff
@@ -88,9 +88,21 @@ pub struct Args {
     /// This skips searching crates.io for a manifest and uses the specified path directly, useful
     /// for debugging and when adding Binstall support. This may be either the path to the folder
     /// containing a Cargo.toml file, or the Cargo.toml file itself.
+    ///
+    /// This option cannot be used with `--git`.
     #[clap(help_heading = "Overrides", long)]
     pub manifest_path: Option<PathBuf>,
 
+    #[cfg(feature = "git")]
+    /// Override how to fetch Cargo.toml package manifest.
+    ///
+    /// This skip searching crates.io and instead clone the repository specified and
+    /// runs as if `--manifest-path $cloned_repo` is passed to binstall.
```

Comment:
Can be for a future improvement, but would be useful to consider supporting some syntax or options for:
- Branches/refs other than default
- Subpaths in the repo

Reply:
I think we can support that by using fragment, e.g.

```
https://github.com/cargo-bins/cargo-binstall#branch=...#subpath=...
```

or maybe using query:

```
https://github.com/cargo-bins/cargo-binstall?branch=...?subpath=...
```

Not sure which one is more compatible with git url, I need more advice on this.

Reply:
I think doing whatever cargo and/or npm does is a good default?

Reply:
For now, [here is the most succinct way](https://github.com/Byron/gitoxide/blob/3cffa268460eb2d41bd6a30d45778b88db4ec602/gix/src/clone/checkout.rs#L67) of checking out a tree into a directory that is fully owned by the caller.
This goes by the big caveat that filters aren't applied and that it will panic on submodules (not implemented) - checking things out is generally not ready yet even though it works well enough for some cases.

On the bright side, I am working on completing the checkout-story next but it will take time to complete - [here is the tracking issue](https://github.com/Byron/gitoxide/issues/301).

Reply:
> For now, [here is the most succinct way](https://github.com/Byron/gitoxide/blob/3cffa268460eb2d41bd6a30d45778b88db4ec602/gix/src/clone/checkout.rs#L67) of checking out a tree into a directory that is fully owned by the caller. This goes by the big caveat that filters aren't applied and that it will panic on submodules (not implemented) - checking things out is generally not ready yet even though it works well enough for some cases.

Thanks, I've already used this for checking out main.
Though I didn't know there's a filter in gix.

> On the bright side, I am working on completing the checkout-story next but it will take time to complete - [here is the tracking issue](https://github.com/Byron/gitoxide/issues/301).

Looks like we would have to wait until then before implementing "--branch".
I might need to take a look at PRs linked with https://github.com/rust-lang/cargo/issues/11813 to see how they accomplish this.

Reply:
It looks like refspec can be used to fetch specific rev/commit, I think it could be used to checkout specific branch/tag.

Reply:
Yes, indeed, the combination of choosing a refspec that fetches the desired branch and checking out its tree afterwards is how `--branch` would be implemented. The question of how to deal with the implementation's limitations would still have to be answered though.

> Though I didn't know there's a filter in gix.

I don't know what `filter` means in this context. Feel free to ignore if everything is clear to you though.

> I might need to take a look at PRs linked with [rust-lang/cargo#11813](https://github.com/rust-lang/cargo/issues/11813) to see how they accomplish this.

`cargo` still performs checkouts using `git2`, and it's what I will be working to change from now on.

Reply:
> Yes, indeed, the combination of choosing a refspec that fetches the desired branch and checking out its tree afterwards is how --branch would be implemented.

Thanks for confirmation of this!

>  This goes by the big caveat that filters aren't applied and that it will panic on submodules (not implemented) - checking things out is generally not ready yet even though it works well enough for some cases.

Well that's unfortunate.
Our goal here is to have the same behavior as `cargo-install` when `--git` is passed.
I'm not sure whether it clones submodules though.

> `cargo` still performs checkouts using `git2`, and it's what I will be working to change from now on.

Thanks for the info, I will keep track of the adoption of `gitoxide` in `cargo`

Reply:
It might be too hard to get everything done in one PR, while `gix` doesn't support checking out specific branch/tag/rev at clone.

I will open a new tracking issue for this.

Reply:
Opened tracking issue #1165 for this.